File: BasicSample/app/src/main/java/com/example/android/persistence/db/dao/CommentDao.java
/*
 * Copyright 2017, The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.android.persistence.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.example.android.persistence.db.entity.CommentEntity;

import java.util.List;
//ORM映射,访问数据库方法
@Dao
public interface CommentDao {
    //comments表中 productId = 传入的:productId 返回 LiveData
    @Query("SELECT * FROM comments where productId = :productId")
    LiveData<List<CommentEntity>> loadComments(int productId);
    //comments表中 productId = 传入的:productId
    @Query("SELECT * FROM comments where productId = :productId")
    List<CommentEntity> loadCommentsSync(int productId);
    //插入数据 主键相同,旧数据会替换新数据
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<CommentEntity> comments);
}

File: BasicSample-kotlin/app/src/main/java/com/kotlin/basicsample_kotlin/db/dao/CommentDao.kt
package com.kotlin.basicsample_kotlin.db.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.kotlin.basicsample_kotlin.db.entity.CommentEntity

@Dao
interface CommentDao {
    @Query("SELECT * FROM comments where productId = :productId")
    fun loadComments(productId:Int): LiveData<List<CommentEntity>>
    @Query("SELECT * FROM comments where productId = :productId")
    fun loadCommentsSync(productId: Int):List<CommentEntity>
    //插入数据 主键相同,旧数据会替换新数据
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(comments: List<CommentEntity>)
}